packages/aws-cdk/lib/context-providers/index.ts

```diff
@@ -0,0 +1,43 @@
+import cxapi = require('@aws-cdk/cx-api');
+import { SDK } from '../api/util/sdk';
+import { debug } from '../logging';
+import { Settings } from '../settings';
+import { AZContextProviderPlugin } from './availability-zones';
+import { HostedZoneContextProviderPlugin } from './hosted-zones';
+import { ContextProviderPlugin } from './provider';
+import { SSMContextProviderPlugin } from './ssm-parameters';
+import { VpcNetworkContextProviderPlugin } from './vpcs';
+
+type ProviderConstructor =  (new (sdk: SDK) => ContextProviderPlugin);
+export type ProviderMap = {[name: string]: ProviderConstructor};
+
+/**
+ * Iterate over the list of missing context values and invoke the appropriate providers from the map to retrieve them
+ */
+export async function provideContextValues(
+  missingValues: { [key: string]: cxapi.MissingContext },
+  projectConfig: Settings,
+  sdk: SDK) {
+  for (const key of Object.keys(missingValues)) {
+    const missingContext = missingValues[key];
+
+    const constructor = availableContextProviders[missingContext.provider];
+    if (!constructor) {
+      // tslint:disable-next-line:max-line-length
+      throw new Error(`Unrecognized context provider name: ${missingContext.provider}. You might need to update the toolkit to match the version of the construct library.`);
+    }
+
+    const provider = new constructor(sdk);
+
+    const value = await provider.getValue(missingContext.props);
+    projectConfig.set(['context', key], value);
+    debug(`Setting "${key}" context to ${JSON.stringify(value)}`);
+  }
+}
+
+const availableContextProviders: ProviderMap = {
+  'availability-zones': AZContextProviderPlugin,
+  'ssm': SSMContextProviderPlugin,
+  'hosted-zone': HostedZoneContextProviderPlugin,
+  [cxapi.VPC_PROVIDER]: VpcNetworkContextProviderPlugin,
```

Comment:
Should we add the others to `cxapi` as well?

Reply:
I suppose. Did not want to do that in order to not overcomplicate the PR.